sdk-setup/src/rpmvalidation

```diff
@@ -0,0 +1,238 @@
+    local target=$1
+    local suite=$2
+    local key=$3
+    local type=$4
+    local ini_files=("${CONFIG_DIRS[@]/%//$suite.ini}")
+    local python=
+    read -d '' -r python <<'EOF'
+import sys, configparser;
+config = configparser.RawConfigParser();
+config.read(sys.argv[3:]);
+sec, val = sys.argv[1].split(".");
+type = sys.argv[2];
+if type == 'boolean':
+    print(config.getboolean(sec,val));
+else:
+    print(config[sec][val]);
+EOF
+    sb2 -t "$OPT_TARGET" python3 -c "$python" "RpmValidationSuite.$key" "$type" "${ini_files[@]}"
+}
+
+suite_boolean() {
+    suite_value "$1" "$2" "$3" boolean
+}
+
+suite_raw() {
+    suite_value "$1" "$2" "$3" raw
+}
+ 
+list_target() {
+    local target=$1
+    local display_target=$2
+    local suite=
+    for suite in "${SUITES[@]}"; do
+        local essential=$(suite_boolean "$target" "$suite" Essential)
+        if [[ $essential == True ]]; then
+            essential="Essential"
+        else
+            essential="Optional"
+        fi
+        local website=$(suite_raw "$target" "$suite" Website)
+        local name=$(suite_raw "$target" "$suite" Name)
+        write ${display_target:+"$target"} "$suite" "$essential" "$website" "$name"
+    done
+}
+
+list_all() {
+    local t=
+    for t in $(sdk-manage target list); do
+        populate_suites $t
+        list_target $t 1
+    done
+}
+
+run_suite() {
+    local suite=$1
+    suite_exec=$(suite_raw "$OPT_TARGET" "$suite" Exec)
+    suite_cmd=${suite_exec//%f/$OPT_RPM_FILE}
+
+    sb2 -t "$OPT_TARGET" "$suite_cmd"
+}
+
+run_suites() {
+    echo "Using validation suites: ${SUITES[@]}"
+    local suite=
+    for suite in "${SUITES[@]}"; do
+        run_suite $suite
+    done
+}
+
+set_defaults()
+{
+    OPT_TARGET=
+    OPT_RPM_FILE=
+    OPT_USAGE=
+    OPT_LIST=
+    OPT_SUITES=()
+}
+
+parse_options() {
+    if [[ $0 == *.sh ]]; then
+        echo "Warning: The name '$0' is deprecated. Use '${0%.sh}' instead" >&2
+    fi
+
+    while [[ $1 ]]; do
+        case $1 in
+            -h | --help)
+                OPT_USAGE=1
+                return
+                ;;
+            -t | --target)
+                shift
+                OPT_TARGET=$1
+                ;;
+            -l | --list-suites)
+                OPT_LIST=1
+                ;;
+            --suites )
+                shift
+                IFS=, read -a OPT_SUITES <<<"$1"
+                ;;
+            *)
+                [[ $OPT_RPM_FILE ]] && error "RPM File" "$1" "Only one RPM file can be specified"
+                OPT_RPM_FILE=$1
+                ;;
+        esac
+        shift
+    done
+
+    if [[ ! $OPT_LIST ]]; then
+        if [[ ! $OPT_RPM_FILE ]]; then
+            OPT_USAGE=1
+        fi
+    fi
+}
+
+main() {
+    set_defaults
+
+    parse_options "$@" || return
+
+    if [[ $OPT_USAGE ]]; then
+        usage
+        return
+    fi
+
+    if [[ $OPT_TARGET ]]; then
+        populate_suites "$OPT_TARGET"
+        if [[ "${#SUITES[@]}" -lt 1 ]]; then
```

Comment:
We should not fail here for the `--list-suites` command, so

`if [[ "${#SUITES[@]}" -lt 1 && ! $OPT_LIST ]]; then`